Tools/autotest/common.py

```diff
@@ -1064,22 +1064,33 @@ def wait_waypoint(self,
 
     def wait_mode(self, mode, timeout=60):
         """Wait for mode to change."""
+        if type(mode) == int or mode.isdigit():
+            return self.wait_mode_number(int(mode), timeout=timeout)
         mode_map = self.mav.mode_mapping()
         if mode_map is None or mode not in mode_map:
             self.progress("Unknown mode '%s'" % mode)
             self.progress("Available modes '%s'" % mode_map.keys())
             raise ErrorException()
         self.progress("Waiting for mode %s" % mode)
+        self.wait_mode_number(mode_map[mode], timeout=timeout, verbose=False)
```

Comment:
I prefer to keep the old way, otherwise we will lost the mode readability into the test ...